rfc/2022/2022-06-15 - Keycloak configuration managed by Terraform.md

```diff
@@ -0,0 +1,113 @@
+# RFC: Keycloak configuration will be managed by Terraform to support automation 
+
+<!--
+The title is what you want comments on. Use the active voice in a future tense.
+Example:
+    - The website will be built using the XZY framework
+-->
+
+* Comment Deadline: `2022-06-15`
+* Team Crew and Name: Platform/Access Management Team
+* Authors:
+  * [Ritesh Patel](https://github.com/iriteshp)
+  * [Kevin Duensing](https://github.com/kjduensing)
+* [Original RFC Pull Request](https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/14)
+
+## Background
+Infrastructure as Code (IaC) is the practice of managing and provisioning infrastructure programmatically instead of doing it manually. Configurations are created as code containing resource specifications, making them easier to create, update, manage and distribute. Several IaC solutions are currently available, but we will implement Terraform as IaC to configure Keycloak resources automatically because it is cloud-agnostic and provides a flexible abstraction of resources and providers.
+
+## Motivation
+1. To make the process of creating/managing Keycloak resources automated so less manual intervention is needed.
+2. To maintain user permissions automatically so human errors can be prevented and permissions can be audited in one place.
+3. To ensure consistency so resource conflicts can be prevented. Terraform maintains one state so developers can reference the same configurations no matter where they are running from.
+
+## Design
+The syntax of Terraform configurations is called HashiCorp Configuration Language (HCL). The HCL config files contain configuration code for the Keycloak provider and resources. The diagram below describes our process for  The below process is a potential solution for automating Terraform to create/modify resources in Keycloak.
+
+1. Pull requests in source control trigger automation with appropriate parameters and the parameters are passed to the config files. We will use commands terraform validate and terraform plan to validate configuration files and see the execution plan of the changes.
+2. We will store state files generated by Terraform terraform.tfstate remotely maintains the state of infrastructure and configuration. 
+3. After validating the execution plan when a pull request is approved, the process will trigger the terraform apply  command to execute proposed actions in the execution plan and apply changes in Keycloak.
```

Comment:
How is the execution plan being validated?

Reply:
When GitHub action runs terraform plan it also writes plan execution output to PR. so, the AMT team can quickly validate and approve PR.